Update cilium-agent ServiceMonitor

nebula-it opened the pull request:
Please ensure your pull request adheres to the following guidelines:

- [x] For first time contributors, read [Submitting a pull request](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#submitting-a-pull-request)
- [x] All code is covered by unit and/or runtime tests where feasible.
- [x] All commits contain a well written commit description including a title,
      description and a `Fixes: #XXX` line if the commit addresses a particular
      GitHub issue.
- [x] If your commit description contains a `Fixes: <commit-id>` tag, then
      please add the commit author[s] as reviewer[s] to this issue.
- [x] All commits are signed off. See the section [Developer’s Certificate of Origin](https://docs.cilium.io/en/stable/contributing/development/contributing_guide/#dev-coo)
- [x] Provide a title or release-note blurb suitable for the release notes.
- [] Are you a user of Cilium? Please add yourself to the [Users doc](https://github.com/cilium/cilium/blob/main/USERS.md)
- [ ] Thanks for contributing!

<!-- Description of change -->

Fixes: #issue-number

nebula-it committed:
helm: cleanup envoy metrics

Remove envoy metrics from cilium-agent when envoy daemonset is enabled

Signed-off-by: Nebula <nebula.it@outlook.com>

gandro commented on install/kubernetes/cilium/templates/cilium-agent/servicemonitor.yaml:
```diff
@@ -22,7 +22,7 @@ metadata:
   selector:
     matchLabels:
-      k8s-app: cilium
+      app.kubernetes.io/name: cilium-agent
```
Why this change? All other ServiceMonitor definitions also use "k8s-app", let's be consistent

nebula-it replied:
Sorry, I messed up the commit message while cleaning up my email. Here is the original commit message for this `Update the label to be more precise as exiting label selects both cilium-agent and hubble-peer services`

gandro replied:
Could you update the commit message so it's correct again? Thanks

nebula-it replied:
Done

gandro reviewed (changes requested):
Thanks for the PR! The Envoy DS change looks good to me, but the label change looks unintentional

nebula-it committed:
helm: update selector for cilium-agent servicemonitor

Update the label to be more precise as exiting label select both cilium-agent and hubble-peer services

Signed-off-by: Nebula <40148908+nebula-it@users.noreply.github.com>
